lib/migrations/migrate/Migrator.js

```diff
@@ -583,12 +602,9 @@ function getNewMigrations(migrationSource, all, completed) {
   });
 }
 
-function checkPromise(logger, migrationPromise, name, commitFn) {
+function checkPromise(logger, migrationPromise, name) {
   if (!migrationPromise || typeof migrationPromise.then !== 'function') {
     logger.warn(`migration ${name} did not return a promise`);
-    if (commitFn) {
-      commitFn();
-    }
```

Comment:
I removed the explicit commit so that we can run the `afterEach` hook in the same transaction (line 527)